### PR TITLE
Tokenizer and dataset

### PR DESCRIPTION
**Pull request by KwiatkowskiML**
This pull request introduces significant changes to support bilingual dataset handling and tokenizer creation for a machine translation pipeline. The key updates include defining constants for special tokens, implementing a new `BilingualDataset` class, and adding utility functions for tokenizer creation and dataset preparation.

### Dataset Handling:
* Added a new `BilingualDataset` class in `dataset.py` to handle bilingual data, including tokenization, padding, and masking for both encoder and decoder inputs. The class ensures compatibility with sequence length requirements and supports SOS, EOS, and PAD tokens.

### Tokenizer Creation:
* Added utility functions in `tokenizer.py` to build or load tokenizers for source and target languages using the `tokenizers` library. The tokenizers are trained with special tokens and saved for reuse.

* Implemented a `get_ds` function in `tokenizer.py` to prepare the dataset, split it into training and validation sets, and create corresponding `DataLoader` objects. It also computes and logs the maximum sentence lengths for debugging purposes.

### Constants:
* Added constants for special tokens (`UNKNOWN_TOKEN`, `PAD_TOKEN`, `EOS_TOKEN`, `SOS_TOKEN`) and the dataset name (`DATASET_NAME`) in `constants.py` to ensure consistency across the codebase.

### CODE REVIEW

File: tokenizer.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        src_ids = tokenizer_src.encode(item['translation'][config['lang_src']]).ids
+        tgt_ids = tokenizer_tgt.encode(item['translation'][config['lang_tgt']]).ids
+        max_len_src = max(max_len_src, len(src_ids))
+        max_len_tgt = max(max_len_tgt, len(tgt_ids))
+
+    print(f'Max length of source sentence: {max_len_src}')
+    print(f'Max length of target sentence: {max_len_tgt}')
+
+    train_dataloader = DataLoader(train_ds, batch_size=config['batch_size'], shuffle=True)
+    val_dataloader = DataLoader(val_ds, batch_size=1, shuffle=True)
```
[the remark this answers is not in the document]
Comment (KKKUBAKKK) [reply tokenizer.py:76]: Wydaje mi się, że to jest obra sugestia

**Review by KKKUBAKKK — APPROVED** [review]
